source/common/network/apple_dns_impl.cc

```diff
@@ -185,10 +186,33 @@ void AppleDnsResolverImpl::PendingResolution::onEventCallback(uint32_t events) {
   }
 }
 
+std::list<DnsResponse>& AppleDnsResolverImpl::PendingResolution::finalAddressList() {
+  switch (dns_lookup_family_) {
+  case DnsLookupFamily::V4Only:
+    return pending_cb_.v4_responses_;
+  case DnsLookupFamily::V6Only:
+    return pending_cb_.v6_responses_;
+  case DnsLookupFamily::Auto:
+    // Per API docs only give v4 if v6 is not available.
+    if (pending_cb_.v6_responses_.empty()) {
+      return pending_cb_.v4_responses_;
+    }
+    return pending_cb_.v6_responses_;
+  case DnsLookupFamily::V4Preferred:
+    // Per API docs only give v6 if v4 is not available.
+    if (pending_cb_.v4_responses_.empty()) {
+      return pending_cb_.v6_responses_;
+    }
+    return pending_cb_.v4_responses_;
+  default:
+    NOT_REACHED_GCOVR_EXCL_LINE;
```

Comment:
nit: move this to outside the switch statement if it's required for compilation.